src/test/scala/org/broadinstitute/dsde/firecloud/service/LibraryApiServiceSpec.scala

```diff
@@ -237,5 +237,34 @@ class LibraryApiServiceSpec extends BaseServiceSpec with LibraryApiService with
         }
       }
     }
+
+    "when searching for ORSP IDs" - {
+      "GET on " + duosConsentOrspIdPath - {
+        "should return a valid consent for '12345'" in {
+          Get(duosConsentOrspIdPath + "12345") ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+            status should equal(OK)
+            val consent = response.entity.asString.parseJson.convertTo[Consent]
+            consent shouldNot equal(None)
+            consent.name should equal("12345")
+          }
+        }
+        "should return a Bad Request error on 'unapproved'" in {
+          Get(duosConsentOrspIdPath + "unapproved") ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+            status should equal(BadRequest)
+          }
+        }
+        "should return a Not Found error on known 'missing'" in {
+          Get(duosConsentOrspIdPath + "missing") ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+            status should equal(NotFound)
+          }
+        }
+        "should return a Not Found error on unknown missing" in {
+          val orspId = randomStringFromCharList(10, ('a' to 'z') ++ ('A' to 'Z') ++ ('0' to '9'))
+          Get(duosConsentOrspIdPath + orspId) ~> dummyUserIdHeaders("1234") ~> sealRoute(libraryRoutes) ~> check {
+            status should equal(NotFound)
+          }
+        }
+      }
+    }
```

Comment:
I think http://stackoverflow.com/a/19441215 has your answer. `MockOntologyDAO.orspIdSearch` throws exceptions outside of a Future. If you modify the mock to be structured like the runtime dao, the exceptions are handled by the caller's `recover*`, something like this:

```
  override def orspIdSearch(userInfo: UserInfo, orspId: String): Future[Option[Consent]] = {
    Future.successful(true) map { x =>
      orspId match {
        case x if x.equals("unapproved") =>
          throw new FireCloudExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "Unapproved"))
        case x if x.equals("missing") =>
          throw new FireCloudExceptionWithErrorReport(ErrorReport(StatusCodes.NotFound, "Not Found"))
        case x if x.equals("12345") =>
          Some(Consent(consentId = "consent-id-12345", name = "12345", translatedUseRestriction = Some("Translation")))
        case _ => None
      }
    }
  }
```
(there's probably a cleaner way to write this, instead of `Future.successful(true) map ...`